rust/parser/mod.rs

```diff
@@ -542,7 +525,7 @@ fn visit_schema_rule(_ctx: Rc<Schema_ruleContext>) -> ParserResult<()> {
 
 fn visit_type_any(ctx: Rc<Type_anyContext>) -> ParserResult<Type> {
     if let Some(var) = ctx.VAR_() {
-        Ok(Type::Variable(get_var(var).into_type()))
+        Ok(Type::Variable(get_var(var)))
```

Comment:
`Type::Variable` should really have been `UnboundVariable` from the start, as it can't really have any constraints in this context.